### PR TITLE
Add script 'reboot-stability-check'

### PR DESCRIPTION
**Pull request by okurz**
Originally from https://progress.opensuse.org/projects/openqav3/wiki/#Best-practices-for-infrastructure-work

Related progress issue: https://progress.opensuse.org/issues/133553

### CODE REVIEW

**Comment by okurz** [comment]
@nicksinger

File: reboot-stability-check [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+start="${start:-"01"}"
+runs="${runs:-"30"}"
+host="${1:?"Need 'host' to check reboot stability for. Can be multiple hosts"}"
+boot_timeout="${boot_timeout:-"600"}"
+ping_timeout="${ping_timeout:-"30"}"
+sleep_time="${sleep_time:-"120"}"
+
+check-one-host() {
+    echo -n "run: $run, $host: ping .. "
+        timeout -k 5 "$boot_timeout" sh -c "until ping -c$ping_timeout $host >/dev/null; do :; done"
```
Comment (kalikiana) [inline reboot-stability-check:12]: Is this picking up the individual `$host` from the loop in `main`?

I would find it clearer to use different variables eg. call it `$hosts` elsewhere

Comment (Martchus) [inline reboot-stability-check:12]: The indentation looks a bit odd.

Comment (okurz) [reply reboot-stability-check]: ok, done

Comment (okurz) [reply reboot-stability-check]: fixed

**Review by perlpunk — APPROVED** [review]

File: reboot-stability-check [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+sleep_time="${sleep_time:-"120"}"
+
+check-one-host() {
+    local host=$1
+    echo -n "run: $run, $host: ping .. "
+    timeout -k 5 "$boot_timeout" sh -c "until ping -c$ping_count $host >/dev/null; do :; done"
+    echo -n "ok, ssh .. "
+    timeout -k 5 "$boot_timeout" sh -c "until nc -z -w 1 $host 22; do :; done"
+    echo -n "ok, uptime/reboot: "
+    echo ssh "$host" "uptime && sudo reboot"
```
Comment (Martchus) [inline reboot-stability-check:17]: Is the `echo` still intended here? It looks like a debugging leftover.

Comment (okurz) [reply reboot-stability-check]: yes, thank you. Fixed

**Review by kalikiana — APPROVED** [review]

**Pull request merged**